trajectorytools/plot/polar.py

```diff
@@ -0,0 +1,176 @@
+import logging
+
+import numpy as np
+import scipy
+
+from collections import namedtuple
+
+__all__ = ["binned_statistic_polar", "polar_histogram", "plot_polar_histogram"]
+
+
+def remove_nans_from_args(f_unwrapped):
+    """ Decorator that removes nans from input numpy arrays
+    """
+
+    def wrapped(*args, **kwargs):
+        for arg in args:
+            if not isinstance(arg, np.ndarray):
+                raise TypeError("Input needs to be np array")
+
+            # Maybe move this to functions?
+            if arg.ndim != 1 or len(arg) != len(args[0]):
+                raise ValueError("Input arrays must be 1d with same length")
+
+        # Remove nans, if any
+        finite_values = np.logical_and.reduce(
+            [np.isfinite(arg) for arg in args]
+        )
+        if not np.all(finite_values):
+            print(f"Keeping only finite values in {f_unwrapped} input")
+            logging.info(f"Keeping only finite values in {f_unwrapped} input")
+            args = [arg[finite_values] for arg in args]
```

Comment:
Could it happen that the lengths of the args is different after removing the non-finite values? Should you check that?

Reply:
No, because if there are nans in one array I am removing these positions from all arrays